services/store.go

```diff
@@ -1050,6 +1030,89 @@ func decodeTransfer(height uint32, info *event.ExecuteNotify, evt *event.NotifyE
 	return xfer
 }
 
+func checkEvmEventLog(evt *event.NotifyEventInfo) (bool, *ctypes.StorageLog) {
+	ethLog, err := event.NotifyEventInfoToEvmLog(evt)
+	if err != nil {
+		return false, nil
+	}
+	return true, ethLog
+}
+func parseEvmOngTransferLog(ethLog *ctypes.StorageLog) (*transfer, error) {
+	ongLog := types2.Log{
+		Address: ethLog.Address,
+		Topics:  ethLog.Topics,
+		Data:    ethLog.Data,
+	}
+	if ongLog.Address.Hex() == ONG_ADDR {
+		parsed, _ := abi.JSON(strings.NewReader(ERC20ABI))
+		nbc := bind.NewBoundContract(ethcom.Address{}, parsed, nil, nil, nil)
+		tf := new(ERC20Transfer)
+		err := nbc.UnpackLog(tf, "Transfer", ongLog)
+		if err != nil {
+			return nil, err
+		}
+		from, err := common.AddressFromHexString(tf.From.Hex())
+		if err != nil {
+			return nil, err
+		}
+		to, err := common.AddressFromHexString(tf.To.Hex())
+		if err != nil {
+			return nil, err
+		}
+		xfer := &transfer{
+			amount: tf.Value,
+			from:   from,
+			to:     to,
+		}
+		if tf.To.Hex() == GOV_ADDR {
+			xfer.isGas = true
+		}
+		return xfer, nil
+	} else {
+		return nil, fmt.Errorf("ont ong transfer")
+	}
+}
+
+func balanceCal(xfer *transfer, evt *event.NotifyEventInfo, diffs map[common.Address]map[common.Address]*big.Int, transfers []*model.Transfer) {
+	if xfer.from != nullAddr {
+		accts, ok := diffs[xfer.from]
+		if ok {
+			balance, ok := accts[evt.ContractAddress]
+			if ok {
+				accts[evt.ContractAddress] = balance.Sub(balance, xfer.amount)
+			} else {
+				accts[evt.ContractAddress] = (&big.Int{}).Neg(xfer.amount)
+			}
+		} else {
+			diffs[xfer.from] = map[common.Address]*big.Int{
+				evt.ContractAddress: (&big.Int{}).Neg(xfer.amount),
+			}
+		}
+	}
+	if xfer.to != nullAddr {
+		accts, ok := diffs[xfer.to]
+		if ok {
+			balance, ok := accts[evt.ContractAddress]
+			if ok {
+				accts[evt.ContractAddress] = balance.Add(balance, xfer.amount)
+			} else {
+				accts[evt.ContractAddress] = xfer.amount
+			}
+		} else {
+			diffs[xfer.to] = map[common.Address]*big.Int{
+				evt.ContractAddress: xfer.amount,
+			}
+		}
+	}
+	transfers = append(transfers, &model.Transfer{
```

Comment:
The `transfers` value needs to be returned from this function, otherwise the appended value could be lost.

Reply:
yes,I fix it